lib/store/index.js

```diff
@@ -76,7 +76,13 @@ export class Store {
       const { start, end } = cell;
       let source = codeManager.getTextInRange(editor, start, end);
       source = source ? source : "";
-      const newCell = commutable.emptyCodeCell.set("source", source);
+      const cellType = codeManager.getMetadataForRow(editor, start);
+      if (cellType === "code") {
+        var newCell = commutable.emptyCodeCell.set("source", source);
+      } else if (cellType === "markdown") {
+        source = codeManager.removeCommentsMarkdownCell(editor, source);
+        var newCell = commutable.emptyMarkdownCell.set("source", source);
+      }
```

Comment:
I'd prefer the use of let here, but that's personal preference:
```js
const cellType = codeManager.getMetadataForRow(editor, start);
let newCell;
if (cellType === "code") {
  newCell = commutable.emptyCodeCell.set("source", source);
} else if (cellType === "markdown") {
  source = codeManager.removeCommentsMarkdownCell(editor, source);
  newCell = commutable.emptyMarkdownCell.set("source", source);
}
```

Reply:
Fixed